x-pack/legacy/plugins/ingest/server/libs/adapters/policy/adapter_types.ts

```diff
@@ -6,106 +6,193 @@
 
 import * as t from 'io-ts';
 
-export const RuntimeDatasourceInput = t.intersection([
-  t.interface(
-    {
-      data_source_id: t.string,
-    },
-    'DatasourceInput'
-  ),
+/**
+ * The entire config for the Beats agent, including all assigned data source config outputs
+ * along with agent-wide configuration values
+ */
+
+export const RuntimeStoredPolicy = t.intersection([
```

Comment:
Split the policy into 2 types. How it is stored as a runtime and static type using io-ts for validation,  and the populated static type.